.github/workflows/test-and-push.yaml

```diff
@@ -22,39 +20,128 @@ jobs:
           pip install yamllint
           yamllint .
 
-  test:
+      - name: Install yq
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y yq
+
+  unit-tests:
     runs-on: ubuntu-latest
     needs: pre-build-checks
     steps:
       - name: Checkout code
         uses: actions/checkout@v4
 
-      - name: Set up Kind cluster
-        uses: helm/kind-action@v1.8.0
+      - name: Set up Go
+        uses: actions/setup-go@v4
         with:
-          version: "v0.20.0"
+          go-version: "1.21"
 
-      - name: Install dependencies
+      - name: Clean Go cache
         run: |
-          sudo apt-get update
-          sudo apt-get install -y bash
+          go clean -cache
+          go clean -modcache
+          sudo rm -rf /usr/local/go/pkg || true
+
+      - name: Run unit tests
+        run: make test-unit
 
-      - name: Run test suite
+      - name: Upload coverage
+        uses: codecov/codecov-action@v3
+        with:
+          file: coverage.out
+          flags: unittests
+
+  integration-tests:
+    runs-on: ubuntu-latest
+    needs: [pre-build-checks, unit-tests]
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v2
+
+      - name: Install Kind and kubectl
         run: |
-          make test
-          make clean
+          # Install Kind
+          curl -Lo ./kind https://kind.sigs.k8s.io/dl/v0.20.0/kind-linux-amd64
+          chmod +x ./kind
+          sudo mv ./kind /usr/local/bin/kind
+
+          # Install kubectl
+          curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+          chmod +x kubectl
+          sudo mv kubectl /usr/local/bin/
+
+      - name: Create Kind cluster with custom config
+        run: |
+          kind create cluster --config - <<EOF
```

Comment:
possibly extract to makefile

Reply:
I removed kind since it was troublesome in github actions and I'm relying on unit testing.